core/store/orm/orm.go

```diff
@@ -137,40 +136,6 @@ func (orm *ORM) Unscoped() *ORM {
 	}
 }
 
-// UpsertNodeVersion inserts a new NodeVersion
-func (orm *ORM) UpsertNodeVersion(version models.NodeVersion) error {
-	if err := orm.MustEnsureAdvisoryLock(); err != nil {
-		return err
-	}
-
-	return orm.Transaction(func(tx *gorm.DB) error {
-		err := tx.Clauses(clause.OnConflict{
-			DoNothing: true,
-		}).Create(&version).Error
-		if err != nil {
-			return err
-		}
-		return nil
-	})
-}
-
-// FindLatestNodeVersion looks up the latest node version
-func (orm *ORM) FindLatestNodeVersion() (*models.NodeVersion, error) {
-	if err := orm.MustEnsureAdvisoryLock(); err != nil {
-		return nil, err
-	}
-	var nodeVersion models.NodeVersion
-	err := orm.DB.Order("created_at DESC").First(&nodeVersion).Error
-	if err == gorm.ErrRecordNotFound {
-		return nil, nil
-	}
-	if err != nil && strings.Contains(err.Error(), "relation \"node_versions\" does not exist") {
-		logger.Default.Debug("Failed to find any node version in the DB, the node_versions table does not exist yet.")
```

Comment:
hm i wonder if we maybe still need to catch this error, as the Upsert might be run before the first migration?

Reply:
Ah ok. Sure

Reply:
I've moved the error check out of the ORM and into the caller. The ORM shouldn't handle use case specific logic like how to deal with errors, and other callers of the function may be expecting errors such a NotFound